src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -181,3 +183,15 @@ interface TriggeredInfo {
     val user: User?
     val build: Build?
 }
+
+interface FinishBuildTrigger {
+    val initiatedBuildConfiguration: BuildConfigurationId
+    val afterSuccessfulBuildOnly: Boolean
+    val branchFilter: List<BranchRule>
+}
+
+interface BranchRule {
+    val rule: String
+    val include: Boolean
+    val name: String
```

Comment:
And BTW according to TeamCity [docs](https://confluence.jetbrains.com/display/TCD10/Configuring+Finish+Build+Trigger) it may be not branch name, but a branch pattern with wildcard inside.